sbin/build.sh

```diff
@@ -792,19 +792,21 @@ getFirstTagFromOpenJDKGitRepo() {
   # prefix "-" to allow line numbering stable sorting using nl => -LLL-BB
   # Sort by build level BB first
   # Then do "stable" sort (keeping BB order) by build level LLL
-  local jdk8_tag_sort1="sort -t- -k3n"
-  local jdk8_tag_sort2="sort -t- -k2n"
+  local jdk8_tag_sort1="sort -t- -k3,3n"
+  local jdk8_tag_sort2="sort -t- -k2,2n"
   local jdk8_get_tag_cmd="grep -v _openj9 | grep -v _adopt | cut -c6- | awk -F'[\-b]+' '{print \$1\"-\"\$2}' | sed 's/^/-/' | $jdk8_tag_sort1 | nl | $jdk8_tag_sort2 | cut -f2- | sed 's/^-/jdk8u/' | sed 's/-/-b/' | tail -1"
 
   # JDK11+ tag sorting:
   # We use sort and tail to choose the latest tag in case more than one refers the same commit.
   # Versions tags are formatted: jdk-V[.W[.X[.P]]]+B; with V, W, X, P, B being numeric.
   # Transform "-" to "." in tag so we can sort as: "jdk.V[.W[.X[.P]]]+B"
+  # Transform "+" to ".0.+" during the sort so that .P (patch) is defaulted to "0" for those
+  # that don't have one, and the trailing "." to terminate the 5th field from the +
   # First, sort on build number (B):
-  local jdk11plus_tag_sort1="sort -t+ -k2n"
-  # Second, (stable) sort on (V), (W), (X):
-  local jdk11plus_tag_sort2="sort -t. -k2n -k3n -k4n -k5n"
-  jdk11plus_get_tag_cmd="grep -v _openj9 | grep -v _adopt | sed 's/jdk-/jdk./g' | $jdk11plus_tag_sort1 | nl | $jdk11plus_tag_sort2 | cut -f2- | sed 's/jdk./jdk-/g' | tail -1"
+  local jdk11plus_tag_sort1="sort -t+ -k2,2n"
+  # Second, (stable) sort on (V), (W), (X), (P): P(Patch) is optional and defaulted to "0"
+  local jdk11plus_tag_sort2="sort -t. -k2,2n -k3,3n -k4,4n -k5,5n"
+  jdk11plus_get_tag_cmd="grep -v _openj9 | grep -v _adopt | sed 's/jdk-/jdk./g' | sed 's/+/.0.+/g' | $jdk11plus_tag_sort1 | nl | $jdk11plus_tag_sort2 | sed 's/.0.+/+/g' | cut -f2- | sed 's/jdk./jdk-/g' | tail -1"
```

Comment:
As mentioned I cannot see where the (existing, not added by this PR) `nl` comes into helping produce a stable sort, but this needs to be changed as `.` is a wildcard character. It works by luck, but best to make it right:
```suggestion
  jdk11plus_get_tag_cmd="grep -v _openj9 | grep -v _adopt | sed 's/jdk-/jdk./g' | sed 's/+/.0.+/g' | $jdk11plus_tag_sort1 | nl | $jdk11plus_tag_sort2 | sed 's/\.0\.+/+/g' | cut -f2- | sed 's/jdk./jdk-/g' | tail -1"
```
The extra `.` after the `0` in these translations is superfluous but doesn't do any harm.

Reply:
i worked it out again (I should have written it down the first time!), and added explanation to the original PR here: https://github.com/AdoptOpenJDK/openjdk-build/issues/2101#issuecomment-728779484

Reply:
the extra "." is needed so that the -t. -k5,5n ensures the 5th "numeric" sort field does not include the "+",eg.
jdk.11.0.9.1+1
the 5th field using "." field separator is "1+1"
using
jdk.11.0.9.1.0.+1
the 5th field is "1"
also note the extra ".0." is superflous in this example, but not when there is no patch eg.
jdk.11.0.8+1 => 4th field is "8+1", 5th is "blank" 
jdk.11.0.8.0.+1 => 4th field is "8", 5th field is "0"

Reply:
)To be clear, my "superfluous" comment related to the final `.` (i.e. the use of `.0.` instead of just `.0`

Reply:
yes, in my example if + was only changed to .0:
jdk.11.0.8+1 => jdk.11.0.8.0+1 => 4th field is "8", 5th field is **"0+"**
**"0+"** won't numerically sort correctly